src/NumericalAlgorithms/DiscontinuousGalerkin/Actions/ApplyBoundaryFluxesLocalTimeStepping.hpp

```diff
@@ -0,0 +1,136 @@
+class ConstGlobalCache;
+}  // namespace Parallel
+// IWYU pragma: no_forward_declare db::DataBox
+/// \endcond
+
+namespace dg {
+namespace Actions {
+/// \ingroup ActionsGroup
+/// \ingroup DiscontinuousGalerkinGroup
+/// \brief Perform the boundary part of the update of the variables
+/// for local time stepping.
+///
+/// Uses:
+/// - ConstGlobalCache:
+///   - CacheTags::TimeStepper
+///   - Metavariables::normal_dot_numerical_flux
+/// - DataBox:
+///   - Tags::Mesh<volume_dim>
+///   - Tags::Mortars<Tags::Mesh<volume_dim - 1>, volume_dim>
+///   - Tags::Mortars<Tags::MortarSize<volume_dim - 1>, volume_dim>
+///   - Tags::TimeStep
+///
+/// DataBox changes:
+/// - Adds: nothing
+/// - Removes: nothing
+/// - Modifies:
+///   - variables_tag
+///   - FluxCommunicationTypes<Metavariables>::mortar_data_tag
+struct ApplyBoundaryFluxesLocalTimeStepping {
+  template <typename DbTags, typename... InboxTags, typename Metavariables,
+            typename ArrayIndex, typename ActionList,
+            typename ParallelComponent>
+  static auto apply(db::DataBox<DbTags>& box,
+                    const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+                    const Parallel::ConstGlobalCache<Metavariables>& cache,
+                    const ArrayIndex& /*array_index*/,
+                    const ActionList /*meta*/,
+                    const ParallelComponent* const /*meta*/) noexcept {
+    using system = typename Metavariables::system;
+    constexpr size_t volume_dim = system::volume_dim;
+    using variables_tag = typename system::variables_tag;
+
+    using flux_comm_types = FluxCommunicationTypes<Metavariables>;
+
+    using mortar_data_tag =
+        typename flux_comm_types::local_time_stepping_mortar_data_tag;
+    db::mutate<variables_tag, mortar_data_tag>(
+        make_not_null(&box),
+        [&cache](
+            const gsl::not_null<db::item_type<variables_tag>*> vars,
+            const gsl::not_null<db::item_type<mortar_data_tag>*> mortar_data,
+            const db::item_type<Tags::Mesh<volume_dim>>& mesh,
+            const db::item_type<Tags::Mortars<Tags::Mesh<volume_dim - 1>,
+                                              volume_dim>>& mortar_meshes,
+            const db::item_type<Tags::Mortars<Tags::MortarSize<volume_dim - 1>,
+                                              volume_dim>>& mortar_sizes,
+            const db::item_type<Tags::TimeStep>& time_step) noexcept {
+          // Having the lambda just wrap another lambda works around a
+          // gcc 6.4.0 segfault.
+          [
+            &cache, &vars, &mortar_data, &mesh, &mortar_meshes, &mortar_sizes,
+            &time_step
+          ]() noexcept {
+            const auto& normal_dot_numerical_flux_computer =
+                get<typename Metavariables::normal_dot_numerical_flux>(cache);
+            const auto& time_stepper = get<CacheTags::TimeStepper>(cache);
+
+            for (auto& this_mortar_data : *mortar_data) {
```

Comment:
this_ -> current_

Reply:
I've reconsidered this since our discussion earlier.  The structures being passed around here are time histories, so I want to avoid words like "current" that have strong time connotations.  Does anyone have another suggestion?

Reply:
that sounds reasonable, my only suggestions are `the_mortar_data` or `a_mortar_data`

Reply:
other suggestions: `mortar_datum`, `mortar_id_and_data` (reflected its use below)

Reply:
> `mortar_id_and_data`

I think we've used this scheme in the past for similar things.  Let's go with it.